src/main/java/com/nha/abdm/wrapper/hip/hrp/link/hipInitiated/HipLinkService.java

```diff
@@ -217,9 +265,17 @@ public void hipConfirmCallOtp(VerifyOTP verifyOTP) {
   public void hipAddCareContext(LinkOnConfirmResponse linkOnConfirmResponse) {
     RequestLog existingRecord =
         logsRepo.findByGatewayRequestId(linkOnConfirmResponse.getResp().getRequestId());
+    if (existingRecord == null) {
+      log.error("hipAddCareContext: Illegal state - Gateway request Id not found in database");
```

Comment:
Returning the error messages to the caller would be good here. Probably throw an exception, something like `InvalidDataException`.

Reply:
The function is subsequent action on gateway call. Created https://github.com/NHA-ABDM/ABDM-wrapper/issues/37 to address this.